### PR TITLE
Seperate question types for each queue

### LINKED ISSUE #95: Seperate Question Types for async

### PR DESCRIPTION
**Pull request by Ferdinand737**
# Description

- Each row in `question_type_model` is now associated with a queue (`null` for async questions) 
- Queues only show the question types that were made for that queue.
- When professor adds question type to queue, the type is private to that queue and cannot be used elsewhere.
- Deleting question types now uses `id` as key instead of `name`
- Updated seeder script

Closes #95

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Integration testing
- [x] manual testing

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code where needed 
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/app/hooks/useQuestionTypes.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
-import { API } from "@koh/api-client";
+import { API } from '@koh/api-client'
 
-export function useQuestionTypes(cid: number): string[] {
-  async () =>
-    API.questions.questionTypes(cid).then((results) => {
+export function useQuestionTypes(cid: number, qid: number): string[] {
```
Comment (Ferdinand737) [inline packages/app/hooks/useQuestionTypes.ts:3]: Just noticed that this hook is not used anywhere

File: packages/api-client/index.ts [hunk 2]
```diff
@@ -386,11 +393,8 @@ class APIClient {
         undefined,
         body,
       ),
-    deleteQuestionType: async (
-      courseId: number,
-      questionType: string,
-    ): Promise<void> =>
-      this.req('DELETE', `/api/v1/questions/${courseId}/${questionType}`),
+    deleteQuestionType: async (questionTypeId: number): Promise<void> =>
```
Comment (d3li0n) [inline packages/api-client/index.ts:396]: Shouldn't we delete questions types by course?

**Review by d3li0n — COMMENTED** [review]
One comment

**Review by AdamFipke — APPROVED** [review]
Tried it out and it works like a charm. Great work!

Things I manually tested:
- creating question types for two different queues
- creating questions using these question types in both queues
- Creating question types for the async queue
- Creating a question using the new question type in the async queue
- searching by question type in async queue

**Review by d3li0n — APPROVED** [review]

**Pull request merged**